patch/cisco-Enable-static-memory-reservation-for-OIRable-PCIe-de.patch

```diff
@@ -0,0 +1,198 @@
++               return size;
++
++       tmp_bus = bus;
++       while (tmp_bus->parent) {
++               tmp_bus = tmp_bus->parent;
++               depth++;
++       }
++
++       /* Only for Second level bridges */
++       if (depth != 5)
++               return size;
++
++       nt_virt_dev = plx_find_nt_device(bus->parent, 0x87b0);
++       if (nt_virt_dev) {
++               pci_read_config_word(nt_virt_dev, PLX_RES_MAGIC_OFFSET,
++                       &res_magic);
++               dev_dbg(&nt_virt_dev->dev,
++                       "Magic offset of 0x%x found in NT device\n", res_magic);
++       }
++
++       if (res_magic == PLX_RES_MAGIC_VALUE) {
++               /*
++                * The pacifics are connected on PLX ports:
++                *  FC4 and FC8: #3, #4
++                *  FC12       : #3, #4, #5
++                *  FC18       : #3, #4, #5, #11
++                */
++
++               /* Calculate resource based on EEPROM values */
++               ds_port_offset = (bus->number - bus->parent->number) - 1;
++               if (ds_port_offset < 5) {
++                       pci_read_config_dword(nt_virt_dev, PLX_RES_DS_PORT_REG0,
++                               &mem_res_bitmap);
++               } else {
++                       ds_port_offset -= 5;
++                       pci_read_config_dword(nt_virt_dev, PLX_RES_DS_PORT_REG1,
++                               &mem_res_bitmap);
++               }
++               port_bitmap = mem_res_bitmap;
++               dev_dbg(&bus->dev, "Port offset: 0x%x, res bitmap 0x%x\n",
++                       ds_port_offset, mem_res_bitmap);
++
++               if (ds_port_offset < 5) {
++                       u8 m[] = { 26, 20, 14, 8, 2 };
++                       u8 s[] = { 31, 25, 19, 13, 7 };
++
++                       multiplier = (port_bitmap >> m[ds_port_offset]) & PLX_RES_P_MASK;
++                       np_size = (port_bitmap >> s[ds_port_offset]) & PLX_RES_NP_MASK;
++
++                       dev_dbg(&bus->dev, "Multiplier: %d, np_size: %d\n",
++                               multiplier, np_size);
++
++                       if (res_type & IORESOURCE_PREFETCH) {
++                               size = 0x100000 << (multiplier - 1);
++                               dev_dbg(&bus->dev, "Pref Multiplier %d, Size 0x%llx\n",
++                                       multiplier, (long long) size);
++                       } else if (np_size) {
++                               size = 0x100000;
++                               dev_dbg(&bus->dev, "NP Size 0x%llx\n", (long long) size);
++                       }
++                }
++       }
++       return size;
++}
++
+ /**
+  * pbus_size_mem() - size the memory window of a given bus
+  *
+@@ -1013,6 +1156,7 @@ static int pbus_size_mem(struct pci_bus *bus, unsigned long mask,
+ 	resource_size_t children_add_size = 0;
+ 	resource_size_t children_add_align = 0;
+ 	resource_size_t add_align = 0;
++        unsigned int dev_count = 0;
```

Comment:
Wrong indentation?

Reply:
Fixed indentation issues and verified running checkpatch.pl